runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -287,5 +310,81 @@ MM_IndexableObjectAllocationModel::layoutDiscontiguousArraylet(MM_EnvironmentBas
 	return spine;
 }
 
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+#if !(defined(LINUX) && defined(J9VM_ENV_DATA64)) || defined(OMRZTPF)
```

Comment:
`|| defined(OMRZTPF)`

I don't think this is necessary as the zTPF platform just won't enable the `J9VM_GC_ENABLE_DOUBLE_MAP` flag